brian2/synapses/synapses.py

```diff
@@ -344,8 +337,8 @@ def before_run(self, run_namespace):
                                                              needed_variables=needed_variables,
                                                              template_kwds=template_kwds,
                                                              run_namespace=run_namespace)
-
-        self._code_objects.insert(0, weakref.proxy(self._pushspikes_codeobj))
+        self.code_objects[:] = [weakref.proxy(self._pushspikes_codeobj),
+                                weakref.proxy(self.create_default_code_object(run_namespace))]
```

Comment:
Didn't check the logic here carefully, but is this always OK to replace ``code_objects`` rather than insert a new item?

Reply:
In this example we are explicitly overwriting the standard logic to sneak in an additional code object, and we cannot create this other object after the standard code object creation has been done. This is all quite convoluted and difficult to follow – definitely a prime candidate for some refactoring. For now, I don't see how to do it differently without doing massive changes.

To make this a bit clearer: in `Synapses.create_code_objects`, we overwrite the default procedure from `CodeRunner.create_code_objects`, where `self.code_objects` is replaced:
https://github.com/brian-team/brian2/blob/bf5f4bc3971c463e470dcad890d2e6c13b6d584e/brian2/groups/group.py#L1132-L1139